core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -1064,12 +1064,7 @@ private void processWorkerAddedBlocks(MasterWorkerInfo workerInfo,
           Optional<BlockMeta> block = mBlockStore.getBlock(blockId);
           if (block.isPresent()) {
             workerInfo.addBlock(blockId);
-            BlockLocation location = entry.getKey();
-            Preconditions.checkState(location.getWorkerId() == workerInfo.getId(),
-                String.format("BlockLocation has a different workerId %s from "
-                    + "the request sender's workerId %s!",
-                        location.getWorkerId(), workerInfo.getId()));
-            mBlockStore.addLocation(blockId, location);
+            mBlockStore.addLocation(blockId, entry.getKey());
```

Comment:
Did you try an `if` statement instead of `Preconditions`? Don't we still need this check?

Reply:
@ggezer @jiacheliu3 Preconditions supports passing string format arguments

```

this can be replaced by

```java
Preconditions.checkState(location.getWorkerId() == workerInfo.getId(),
    "BlockLocation has a different workerId %s from the request sender's workerId %s!",
    location.getWorkerId(), workerInfo.getId());
```

Reply:
Passing string format arguments is proven to save the performance the same way. Although `Preconditions.checkState` is effectively `if(!condition) throw new IllegalStateException`, I think it makes sense to use `Preconditions` instead of manually managing if-else.

Updated according to Bin's advice, thanks!